pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -118,146 +127,170 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 		Password(string(secret.Data["password"])).
-		Insecure(true).
+		Insecure(insecure).
 		Build()
 }
 
 // newDVForCR returns a VM specification of the fetched oVirt VM
-func newDVForCR(vm *ovirtsdk.Vm, secret *corev1.Secret) []cdiv1.DataVolume {
-	quantity, _ := resource.ParseQuantity("1Gi")
-	diskAttachments, _ := vm.DiskAttachments()
+func newDVForCR(diskAttachments *ovirtsdk.DiskAttachmentSlice, secret *corev1.Secret) []cdiv1.DataVolume {
+	quantity, _ := resource.ParseQuantity("1Gi") // FIXME:
 	dvs := make([]cdiv1.DataVolume, len(diskAttachments.Slice()))
-	for _, diskAttachment := range diskAttachments.Slice() {
+	for i, diskAttachment := range diskAttachments.Slice() {
 		disk, _ := diskAttachment.Disk()
-		dvs = append(dvs,
-			cdiv1.DataVolume{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      disk.MustId(), // FIXME:
-					Namespace: "default",     // FIXME:
-					Labels: map[string]string{ // FIXME:
-						"origin": "oVirt",
+		dvs[i] = cdiv1.DataVolume{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "cdi.kubevirt.io/v1alpha1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      diskAttachment.MustId(), // FIXME:
+				Namespace: "default",               // FIXME:
+				Labels: map[string]string{ // FIXME:
+					"origin": "oVirt",
+				},
+			},
+			Spec: cdiv1.DataVolumeSpec{
+				Source: cdiv1.DataVolumeSource{
+					Imageio: &cdiv1.DataVolumeSourceImageIO{
+						URL:           string(secret.Data["apiUrl"]),
+						DiskID:        disk.MustId(),
+						SecretRef:     "ovirt-key", // FIXME, should be created dynamically
+						CertConfigMap: "ovirt-ca",  // FIXME, should be created dynamically
 					},
 				},
-				Spec: cdiv1.DataVolumeSpec{
-					Source: cdiv1.DataVolumeSource{
-						/*
-							Imageio: &cdiv1.DataVolumeSourceImageIO{
-								URL:           secret.Data["apiUrl"],
-								DiskID:        vm.MustDiskAttachments().Slice()[0].MustDisk().MustId(),
-								SecretRef:     "",
-								CertConfigMap: "",
-							},
-						*/
+				// TODO: Would be great to add storageClassName which we should get from the mapping.
+				PVC: &corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
 					},
-					PVC: &corev1.PersistentVolumeClaimSpec{
-						AccessModes: []corev1.PersistentVolumeAccessMode{
-							corev1.ReadWriteOnce,
-						},
-						Resources: corev1.ResourceRequirements{
-							Requests: corev1.ResourceList{
-								corev1.ResourceStorage: quantity,
-							},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: quantity,
 						},
 					},
 				},
 			},
-		)
+		}
 	}
 	return dvs
 }
 
 // newVMForCR returns a VM specification of the fetched oVirt VM
 func newVMForCR(vm *ovirtsdk.Vm, dvs []cdiv1.DataVolume, con *ovirtsdk.Connection) *kubevirtv1.VirtualMachine {
+	// Labels definition:
 	labels := map[string]string{
 		"origin": "oVirt",
```

Comment:
pls reuse constant from previous comment for `oVirt`

Reply:
ok